src/test/java/com/unb/tracker/TrackerApplicationTests.java

```diff
@@ -32,13 +42,75 @@ public void contextLoads() throws Exception {
 
     @Test
     public void shouldReturnIndex() throws Exception {
-        this.mockMvc.perform(get("/")).andDo(print()).andExpect(status().isOk())
+        this.mockMvc.perform(get("/"))
+                .andDo(print())
+                .andExpect(status().isOk())
                 .andExpect(view().name("index"));
     }
 
     @Test
     public void shouldHit404() throws Exception {
-        this.mockMvc.perform(get("/thisdoesnotexist")).andDo(print()).andExpect(status().is(404));
+        this.mockMvc.perform(get("/thisdoesnotexist"))
+                .andDo(print())
+                .andExpect(status().is(404));
+    }
+
+    @Test
+    public void loadCourseFormView() throws Exception {
+        this.mockMvc.perform(get("/course"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(view().name("courseFormView"));
+    }
+    @Test
+    public void saveCourse() throws Exception {
+        String name = "name";
+        String timeSlot = "8:30";
+        String startDate = "2017-01-01";
+        String endDate = "2017-01-01";
+        Course myCourse = new Course();
+        myCourse.setName(name);
+
+        myCourse.setStartDate(convertToSqlDate(startDate));
+        myCourse.setEndDate(convertToSqlDate(endDate));
+        myCourse.setTimeSlot(timeSlot);
+        this.mockMvc.perform(post("/course")
+                .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+                .param("name", name)
+                .param("timeSlot", timeSlot)
+                .param("startDate", startDate)
+                .param("endDate", endDate))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(view().name("courseFormView"));
+
+        Iterable<Course> courses = courseRepository.findAll();
+        assertThat(Iterables.contains(courses, myCourse));
+    }
+
+    private Date convertToSqlDate(String dateString) throws Exception {
+            return new java.sql.Date(dateFormat.parse(dateString).getTime());
+    }
+
+    @Test
+    public void hitCourseValidation() throws Exception {
+        String name = "name";
+        String timeSlot = "8:30";
+        String startDate = "this isn't a real date";
+        String endDate = "2017-01-01";
+        //Dummy placeholder string to solve ambiguous argument problem
+        String s = null;
+        this.mockMvc.perform(post("/course")
+                .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+                .param("name", name)
+                .param("timeSlot", timeSlot)
+                .param("startDate", startDate)
+                .param("endDate", endDate))
+                .andDo(print())
+                .andExpect(status().is(400));
+                //TODO: Add this back once we have validation setup correctly
+                //.andExpect(model().attributeHasErrors("startDate"));
```

Comment:
I realized after I pushed this that this field might not really need form validation as long as we are using the mdbootstrap component everywhere, because unless someone is fucking around with the hidden fields (in which case they deserve to see an error) this shouldn't happen. We will be adding maxlength on the name field, however so there will be an example of that here.